### PR TITLE
[CSFix] Call a function/member which is incorrectly used as a property

### PR DESCRIPTION
**Pull request by xedin**
If the base type for the member lookup is a function, let's try to
use its result type as a base instead, maybe there is just a call
missing.

### CODE REVIEW

**Comment by xedin** [comment]
These changes are prelude to being able to diagnose missing member references via fixes.

**Comment by xedin** [comment]
@swift-ci please test

**Comment by xedin** [comment]
@swift-ci please test source compatibility

File: lib/Sema/CSDiag.cpp [hunk 1]
```diff
@@ -7392,26 +7392,9 @@ bool FailureDiagnosis::diagnoseMemberFailures(
...
-              .fixItInsertAfter(insertLoc, "()");
-          return true;
-        }
-
-      diagnose(baseExpr->getLoc(), diag::did_not_call_function_value)
-          .fixItInsertAfter(insertLoc, "()");
-      return true;
+      auto failure =
+          MissingCallFailure(expr, CS, CS.getConstraintLocator(baseExpr));
+      return failure.diagnoseAsError();
```
Comment (rudkx) [inline lib/Sema/CSDiag.cpp]: Is there a point at which you anticipate this going away in favor of only diagnosing this through fixes?

Comment (xedin) [reply lib/Sema/CSDiag.cpp]: Yes! Are slowly getting there.

**Review by rudkx — APPROVED** [review]

**Pull request merged**